### PR TITLE
BGV Relin Bug Fixes

### PR DESCRIPTION
**Pull request by christopherngutierrez**
## Proposed changes
I fixed a few issues in relin. Namely, the relin kernel generated code that was segmented due issues with composing DigitalDecompExtend and Keymul together. The propose fix uses internal temporary variables output_tmp , which we should refactor out in the future. Further, I also hardcoded input polynomials parts to 3 and input parts to 2 since the code assumes those sizes (also to be refactored to be more generic in the future).

I tested the code by verifying that the output matched our SEAL/legacy implementation. 

## Types of changes

What types of changes does your code introduce to the HE Toolkit project?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you are unsure about any of them, do not hesitate to ask. We are
here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/IntelLabs/hec-p-isa-tools/blob/main/CONTRIBUTING.md) agreement
- [x] Current formatting and unit tests / base functionality passes locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by jobottle — APPROVED** [review]
LGTM, confirmed passing validation tests

**Review by kylanerace — DISMISSED** [review]
LGTM, passing as well

**Comment by faberga** [comment]
Hi @christopherngutierrez ,

Two comments:

I have some questions about the python constructs used in relin.py, particularly storing the "opts" or lists in a variable just to return them at the end of the code block. Since the code block already builds a list, it could be returned directly.

Since you already stated the need for refactoring, and this PR seems to be passing the validation tests, why not use this as your base and refactor now?

-Flavio

File: kerngen/pisa_generators/relin.py [hunk 1]
```diff
@@ -134,3 +145,4 @@ def to_pisa(self) -> list[PIsaOp]:
             Add(self.context, self.output, mul_by_rlk_modded_down, add_original),
             Comment("End of relin kernel"),
         )
+        return opts
```
Comment (faberga) [inline kerngen/pisa_generators/relin.py:148]: See my comments in the PR body.

File: kerngen/kerngen.py
Comment (faberga) [inline kerngen/kerngen.py:1]: HI @christopherngutierrez . Why an empty file?

Comment (kylanerace) [reply kerngen/kerngen.py:1]: Looks moreso to me that the file is now executable. Not empty, just a file permission change (can be run as both "python3 kerngen.py" or "./kerngen.py").

Comment (christopherngutierrez) [reply kerngen/kerngen.py:1]: Kylan answered it correctly. Looks like a prior commit removed the executable permission on the script.

File: kerngen/pisa_generators/relin.py (hunk 1 above)
Comment (christopherngutierrez) [reply kerngen/pisa_generators/relin.py:148]: We can return directly without storing the opts variable. I will modify the code.

**Review by kylanerace — APPROVED** [review]
LGTM, Ran E2E for me with recent change

**Review by faberga — APPROVED** [review]

**Pull request merged**